backend/integration_test/elasticsearch/cluster_test.go

```diff
@@ -129,3 +130,108 @@ func TestSpanClusterUpdates(t *testing.T) {
 		assert.Equal(t, "test", output[0].ObjectId)
 	})
 }
+
+func TestLogCluster(t *testing.T) {
+	if es == nil {
+		t.Error("es is uninitialized or otherwise nil")
+	}
+	ac := client.NewAugurClientImpl(es, client.Immediate)
+	cls := clusterService.NewClusterService(ac, logger)
+	t.Run("should be able to process and update logs of the same type", func(t *testing.T) {
+		err := deleteAllDocuments(es)
+		if err != nil {
+			t.Errorf("Failed to delete all documents: %v", err)
+		}
+		const notAssigned = "NOT_ASSIGNED"
+		onlyTimeStamp := time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC)
+		createdAt := time.Date(1992, 1, 1, 0, 0, 0, 0, time.UTC)
+		const message = "this is a log message"
+		logBatch := []logModel.LogEntry{
+			{
+				ClusterId: notAssigned,
+				CreatedAt: createdAt.Add(-time.Second * 500),
+				Timestamp: onlyTimeStamp,
+				Message:   message,
+			},
+			{
+				ClusterId: notAssigned,
+				CreatedAt: createdAt.Add(-time.Second * 400),
+				Timestamp: onlyTimeStamp,
+				Message:   message,
+			},
+		}
+		err = loadDataIntoElasticsearch(ac, logBatch, bootstrapper.LogIndexName)
+		if err != nil {
+			t.Errorf("Failed to load test data: %v", err)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		input := clusterModel.ClusterInput{
+			DataType:    clusterModel.LogClusterInputType,
+			TextualData: message,
+			ClusterId:   notAssigned,
+			Id:          "test",
+		}
+		output, err := cls.ClusterData(ctx, input)
+		if err != nil {
+			t.Errorf("Failed to cluster span with error: %v", err)
+		}
+		assert.NotZero(t, len(output))
+		assert.Equal(t, 3, len(output))
+		hasTest := output[0].ObjectId == "test"
+		firstClusterId := output[0].ClusterId
+		firstId := output[0].ObjectId
+		for _, cluster := range output[1:] {
+			if cluster.ObjectId == "test" {
+				hasTest = true
+			}
+			assert.Equal(t, firstClusterId, cluster.ClusterId)
+			assert.NotEqual(t, firstId, cluster.ObjectId)
+		}
+		assert.True(t, hasTest)
+	})
+
+	t.Run("should be discriminatory for log messages", func(t *testing.T) {
+		err := deleteAllDocuments(es)
+		if err != nil {
+			t.Errorf("Failed to delete all documents: %v", err)
+		}
+		const notAssigned = "NOT_ASSIGNED"
+		onlyTimeStamp := time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC)
+		createdAt := time.Date(1992, 1, 1, 0, 0, 0, 0, time.UTC)
+		const firstMessage = "Login request successful with username: fake_username and password: fake_password"
+		const nonMatchingMessage = "Login request received with URL /accounts/login and method POST"
```

Comment:
This was my main test to base my changes on. This should **not** match, but it was with the earlier more_like_this_query!